File: ZooProjectJava/src/com/zoo/Zoo.java
package com.zoo;
import java.util.ArrayList;

/**
 * The Zoo class keeps track of its animals.
 * @author lucaspoon
 *
 */
public class Zoo {
	
	// List of all animals in the zoo.
	private ArrayList<Animal> animals;
	
	/**
	 * The default constructor for Zoo.
	 * Initializes the animals list.
	 */
	public Zoo() {
		this.animals = new ArrayList<Animal>();
	}
	
	/**
	 * Adds an animal to the zoo animal list.
	 * @param animal to add to zoo.
	 */
	protected void addAnimal(Animal animal) {
		this.animals.add(animal);
	}
	
	/**
	 * Gets a list of all animals that are in the zoo.
	 * @return the list of animals.
	 */
	protected ArrayList<Animal> getAllAnimals(){
		return this.animals;
	}
	
	/**
	 * Gets the animal that is specified by the id.
	 * @param id This is the id of the animal which corresponds to the index in the animal list.
	 * @return the animal specified in the array list.
	 */
	protected Animal getAnimalById(int id) {
		return this.animals.get(id);
	}
	
	/**
	 * Prints all of the animals in the Zoo.
	 */
	protected void printAnimals() {
		System.out.println("\nAnimals in the Zoo:");
		int i = 0;
		for(Animal animal : this.animals) {
			System.out.println("ID: " + i + ", " + animal.toString());
			i++;
		}
	}
	
	/**
	 * Gets a total count of all the animals in the zoo.
	 * @return the count of animals in the zoo.
	 */
	protected int getAnimalCount() {
		return this.animals.size();
	}

}

File: ZooProjectCpp/src/Zoo.cpp
/*
 * Zoo.cpp
 *
 *  Created on: Mar 31, 2019
 *      Author: lucaspoon
 */

#include "Zoo.h"
#include <iostream>
using namespace std;

Zoo::Zoo() {
	_animals.clear();
}

void Zoo::addAnimal(Animal* animal) {
	_animals.push_back(animal);
}

vector< Animal* > Zoo::getAllAnimals(){
	return _animals;
}

Animal* Zoo::getAnimalById(int animalId) {
	return _animals.at(animalId);
}

void Zoo::printAnimals() {
	cout << "\nAnimals in the Zoo:" << endl;
	int i = 0;
	for(Animal* animal : _animals) {
		cout << "ID: " << i << ", " << *animal << endl;
		i++;
	}
}

int Zoo::getAnimalCount() {
	return _animals.size();
}

Zoo::~Zoo() {
	for(Animal* animal : _animals) {
		delete animal;
	}
	_animals.clear();
}

